libs/math/benchmark/trigonometry/trigonometry.cpp

```diff
@@ -32,7 +32,7 @@ void BM_Sin(benchmark::State &state)
   for (auto _ : state)
   {
     state.PauseTiming();
-    val = T(lfg.AsDouble());
+    val = T(lfg.AsFP<double>());
```

Comment:
These should be replaced with val = lfg.AsType<Type>();

I know that this is not the line being benchmarked, but since this provide a very bad example that other developers might see and copy thinking that it is safe. this is NOT safe!